From 609ca98055e1a62b2e9578804dfe8d772ccd65f5 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:14:03 -0400
Subject: [cpufreq] powernv: Fix build failure on UP

Message-id: <1410545655-205645-615-git-send-email-dzickus@redhat.com>
Patchwork-id: 94578
O-Subject: [RHEL7 PATCH 614/626] cpufreq, powernv: Fix build failure on UP
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit f3cae355a962784101478504ef7f6a389ad62979
Author: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
Date:   Wed Apr 16 11:35:38 2014 +0530

    cpufreq, powernv: Fix build failure on UP

    Paul Gortmaker reported the following build failure of the powernv cpufreq
    driver on UP configs:

    drivers/cpufreq/powernv-cpufreq.c:241:2: error: implicit declaration of
    function 'cpu_sibling_mask' [-Werror=implicit-function-declaration]
    cc1: some warnings being treated as errors
    make[3]: *** [drivers/cpufreq/powernv-cpufreq.o] Error 1
    make[2]: *** [drivers/cpufreq] Error 2
    make[1]: *** [drivers] Error 2
    make: *** [sub-make] Error 2

    The trouble here is that cpu_sibling_mask is defined only in <asm/smp.h>,
    and <linux/smp.h> includes <asm/smp.h> only in SMP builds.

    So fix this build failure by explicitly including <asm/smp.h> in the driver,
    so that we get the definition of cpu_sibling_mask even in UP configurations.

    Reported-by: Paul Gortmaker <paul.gortmaker@windriver.com>
    Signed-off-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/powernv-cpufreq.c b/drivers/cpufreq/powernv-cpufreq.c
index 9edccc6..af49688 100644
--- a/drivers/cpufreq/powernv-cpufreq.c
+++ b/drivers/cpufreq/powernv-cpufreq.c
@@ -29,6 +29,7 @@
 
 #include <asm/cputhreads.h>
 #include <asm/reg.h>
+#include <asm/smp.h> /* Required for cpu_sibling_mask() in UP configs */
 
 #define POWERNV_MAX_PSTATES 256
 
-- 
1.7.1